src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidInstallCommand.cs

```diff
@@ -0,0 +1,116 @@
+        protected override string CommandDescription { get; } = @$"
+{CommandHelp}
+ 
+Arguments:
+";
+
+        public AndroidInstallCommand() : base("install", false, CommandHelp)
+        {
+        }
+
+        protected override Task<ExitCode> InvokeInternal(ILogger logger)
+        {
+            logger.LogDebug($"Android Install command called: App = {_arguments.AppPackagePath}");
+            logger.LogDebug($"Timeout = {_arguments.Timeout.TotalSeconds} seconds.");
+
+            if (!File.Exists(_arguments.AppPackagePath))
+            {
+                logger.LogCritical($"Couldn't find {_arguments.AppPackagePath}!");
+                return Task.FromResult(ExitCode.PACKAGE_NOT_FOUND);
+            }
+
+            // Assumption: APKs we test will only have one arch for now
+            string apkRequiredArchitecture;
+
+            if (string.IsNullOrEmpty(_arguments.DeviceArchitecture))
+            {
+                apkRequiredArchitecture = ApkHelper.GetApkSupportedArchitectures(_arguments.AppPackagePath).First();
+                logger.LogInformation($"Will attempt to run device on detected architecture: '{apkRequiredArchitecture}'");
+            }
+            else
+            {
+                apkRequiredArchitecture = _arguments.DeviceArchitecture;
+                logger.LogInformation($"Will attempt to run device on specified architecture: '{apkRequiredArchitecture}'");
+            }
+
+            var runner = new AdbRunner(logger);
+
+            // Package Name is not guaranteed to match file name, so it needs to be mandatory.
+            return Task.FromResult(InvokeHelper(logger, _arguments.PackageName, _arguments.AppPackagePath, apkRequiredArchitecture, _arguments.DeviceId, runner));
+        }
+
+        public ExitCode InvokeHelper(ILogger logger, string apkPackageName, string appPackagePath, string apkRequiredArchitecture, string? deviceId, AdbRunner runner)
+        {
+            try
+            {
+                using (logger.BeginScope("Initialization and setup of APK on device"))
+                {
+                    // Make sure the adb server is started
+                    runner.StartAdbServer();
+
+                    // enumerate the devices attached and their architectures
+                    // Tell ADB to only use that one (will always use the present one for systems w/ only 1 machine)
+                    var deviceToUse = deviceId ?? runner.GetDeviceToUse(logger, apkRequiredArchitecture, "architecture");
+
+                    if (deviceToUse == null)
+                    {
+                        return ExitCode.ADB_DEVICE_ENUMERATION_FAILURE;
+                    }
+
+                    runner.SetActiveDevice(deviceToUse);
+
+                    // Wait till at least device(s) are ready
+                    runner.WaitForDevice();
+
+                    // Empty log as we'll be uploading the full logcat for this execution
+                    runner.ClearAdbLog();
```

Comment:
I think we probably don't need to clear the adb log on install either, it doesn't seem that useful to me. It's only really important for run because we want to get the test output and not the noise around it.